### DISCUSSION TITLE
ScrollView with Expander staying in view

### DISCUSSION THREAD

**Discussion by tpitman** [xamarin/XamarinCommunityToolkit#1406] [Q&A] [answered]
I have a menu in a ScrollView made up of Expanders. Can someone point me to a way to make sure that when the expander is expanded  by the user if it is outside of the scroll area it gets scrolled into view?

I did find a stack overflow posting, but it was Windows wpf specific and references things that don't seem to existin Xamarin iOS / Android.

**Comment by AndreiMisiukevich** [comment, marked as answer]
@tpitman you always can scroll on your own :)

```csharp
            var exp = new Expander();
            var scr = new ScrollView();

            exp.PropertyChanged += (s, e) =>
            {
                if (e.PropertyName == Expander.StateProperty.PropertyName && exp.State == ExpandState.Expanded)
                {
                    scr.ScrollToAsync(exp, ScrollToPosition.MakeVisible, true);
                }
            };
```

**Reply by tpitman** [reply]
Excellent! Is there a way to do this in xaml?

**Reply by AndreiMisiukevich** [reply]
Nope, unfortunately you cannot call "Scroll" method from xaml( you will have to go to code behind

**Reply by tpitman** [reply]
Ok, I was able to get it working by adding the PropertyChanged="" in the xaml and then a common code behind method that all the expanders can call. Then the sender can be cast to the Expander object and I just needed to name my ScrollView with a name to use for that. Thank you for your help.
